flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/nodes/datastream/DataStreamGroupWindowAggregate.scala

```diff
@@ -148,6 +142,20 @@ class DataStreamGroupWindowAggregate(
         "state size. You may specify a retention time of 0 to not clean up the state.")
     }
 
+    val timestampedInput = if (isRowtimeAttribute(window.timeAttribute)) {
+      // copy the window rowtime attribute into the StreamRecord timestamp field
+      val timeAttribute = window.timeAttribute.asInstanceOf[ResolvedFieldReference].name
+      val timeIdx = inputSchema.logicalFieldNames.indexOf(timeAttribute)
```

Comment:
Yes, in fact we can only do it like this because logical and physical type are the same. Otherwise, we would need to adjust the index.

I'll clean up `RowSchema` a bit more to remove the distinction of logical and physical types.